BE/api/users.js

```diff
@@ -122,8 +123,16 @@ usersRouter.patch("/me", requireUser, async (req, res, next) => {
 usersRouter.get("/:userId/", async (req, res, next) => {
   const { userId } = req.params;
   try {
-    const user = await getUserPageById(userId);
+    let user;
```

Comment:
I think my below code will work a bit better and won't always require you to have an authorization header. 

```js
usersRouter.get("/:userId/", async (req, res, next) => {
  const { userId } = req.params;
  const { user } = req;

  let foundUser;
  try {
    if (user && user.admin) {
      foundUser = await getUserPageById(userId);
    } else {
      foundUser = await getPublicUserPageById(userId);
    }

    res.send(foundUser);
  } catch (err) {
    next(err);
  }
});
```